src/libcadet/model/binding/GeneralizedIonExchangeBinding.cpp

```diff
@@ -84,20 +163,55 @@ inline bool GIEXParamHandler::validateConfig(unsigned int nComp, unsigned int co
 	if (_kD.size() < nComp)
 		throw InvalidParameterException("GIEX_KD requires NCOMP entries");
 	if (_nu.size() < nComp)
-		throw InvalidParameterException("GIEX_NU requires NCOMP entries");
+		throw InvalidParameterException("GIEX_NU requires at least NCOMP entries");
 	if (_sigma.size() < nComp)
 		throw InvalidParameterException("GIEX_SIGMA requires NCOMP entries");
 
+	if (_nu.size() != _nuLin.size())
+		throw InvalidParameterException("GIEX_NU and GIEX_NU_LIN do not have the same length");
+
+	if (_nu.size() != _nuQuad.size())
+		throw InvalidParameterException("GIEX_NU and GIEX_NU_QUAD do not have the same length");
+
+	if (_nu.size() != _nuCube.size())
+		throw InvalidParameterException("GIEX_NU and GIEX_NU_CUBE do not have the same length");
+
+	if ((_nu.size() % nComp) != 0)
+		throw InvalidParameterException("Length of GIEX_NU must be a multiple of NCOMP");
+
+	if ((_nu.size() + nComp != _nuBreaks.size()) && (_nuBreaks.size() > 0))
+		throw InvalidParameterException("GIEX_NU_BREAKS must have one entry more than polynomial pieces in GIEX_NU");
+
+	if ((_nu.size() != nComp) && (_nuBreaks.size() == 0))
+		throw InvalidParameterException("GIEX_NU is expected to have NCOMP entries");
+
 	// Assume monovalent salt ions by default
-	if (_nu.get()[0] <= 0.0)
-		_nu.get()[0] = 1.0;
+	const int nPieces = _nu.size() / nComp;
+	for (int i = 0; i < nPieces; ++i)
+	{
+		if (_nu.get()[i] <= 0.0)
+			_nu.get()[i] = 1.0;
+	}
+
+	// Check breaks
+	for (int i = 0; i < nComp; ++i)
+	{
+		cadet::active const* const b = _nuBreaks.get().data() + nPieces * i;
```

Comment:
Since there are nPieces + 1 breaks for each component, I think this line should read
```
cadet::active const* const b = _nuBreaks.get().data() + (nPieces + 1) * i;
```

Reply:
Well spotted!